### DISCUSSION TITLE
Locatie gekoppeld aan BGT object en het bijwerken daarvan

### DISCUSSION THREAD

**Discussion by marcvanandel** [kadaster-labs/sensrnet-home#202] [Ideas]
Stel dat een sensor aan een topografisch object gekoppeld is voor de locatie, wat moet er dan gebeuren als dat `baseObjectID` object verplaatst wordt? En hoe detecteren we dat?

In de [documentatie](https://kadaster-labs.github.io/sensrnet-home/Model/#views) staat dat er op dit moment geen geautomatiseerde koppeling is tussen (bijv.) BGT en SensRNet. Op welke manier zou dat wenselijk zijn?

1. Idee nr.1 is om een 'vlaggetje' te maken met `volgtBaseObjectID` waarbij de locatie in SensRNet bijgewerkt moet worden (en dus deze koppeling gerealiseerd)
2. Idee nr.2 is om een geautomatiseerde rapportage op te zetten en terugmelding naar bronhouders te doen van grote verschillen tussen locaties van devices en (bijv) BGT objecten

... zijn er nog andere opties, suggesties of behoeften?

**Comment by kad-bloemy** [comment]
Vraagjes:
1. Waarom zou je een sensor gekoppeld willen hebben aan een BGT-object, tenzij om mee te verhuizen als het object verhuist?
2. Als de sensor niet moet meeverhuizen dan volstaat wellicht de mogelijkheid om naar een locatie op de BGT-kaart te linken middels de coördinaten en dus geen object-id te registreren. 
3. Het valt te overwegen om óf de koppeling óf de coördinaten te laten invullen, om ambigue situaties te vermijden.

**Reply by mikegroels** [reply]
1. Omdat je wil weten dat een sensor vast zit aan een inrichtingselement (wat wel wat extra info geeft).
1. Kan een object/inrichtingselement wel verhuizen? Of wordt het dan een nieuw object?

**Comment by CelineJansen** [comment]
3. Van Yolanda vind ik een goede overweging.
Verdwijnende objecten in plaats van objecten die verplaatsen zijn sowieso goed om rekening mee te houden. Verdwijnt het BGT object waaraan de sensor gekoppeld is, dan de sensor automatisch ook verwijderen lijkt me logisch.

**Reply by marcvanandel** [reply]
Als het sensorenregister verder doorontwikkeld is en ook processen, bijv. met de BGT, beter verbonden zijn, dan zou dat kunnen. Op dit moment is die verbondenheid en dus invloed van andere registraties en processen er nog niet ... en daarom vind ik dit een mooi idee voor de toekomst ... niet voor nu.

Voor de korte termijn hebben we vooral de locatie nodig ... en zou ik de koppeling naar de BGT als extra willen beschouwen. Dat geeft de mogelijkheid om een sensor wel op de kaart te tonen en om vervolgens allerlei 'slimme dingen' te ontwikkelen om die koppeling verder uit te werken. Een iets meer pragmatische en bottom-up benadering 😉

**Reply by CelineJansen** [reply]
Wat al zou helpen is de BGT er achter hangen. Dan hoeft er nog geen expliciete koppeling mogelijk te zijn, maar kunnen gebruikers deze impliciet voorlopig wel al zelf maken. Nu is de BGT zelfs niet beschikbaar als achtergrond, waardoor een gebruiker de sensor niet eenvoudig kan plaatsen op een BGT object door simpelweg te kijken waar dit object op de kaart staat.

**Reply by marcvanandel** [reply]
Check. Dat is een vrij gemakkelijk issue: https://github.com/kadaster-labs/sensrnet-registry-frontend/issues/183 😉
